### ISSUE TITLE
Export default and syntax definitions

### ISSUE THREAD

**Issue by disnet** [sweet-js/sweet.js#620]
I just started working on default exports of macros and realized we don't have a form for `syntax/syntaxrec` that would allow this. The naive thing isn't correct:

```js
// this is wrong...
export default syntax m = ctx => ...
// because this is wrong
export default let x = ...
```

The most obvious thing to do is introduce an expression form of `syntax`:

```js
export default syntax ctx => {
  ...
}
```

But that looks weird. Maybe a form that looks like a function declaration?

```js
export default syntax (ctx) {

}
```

Actually, maybe that's nice just in general. We could drop the `syntaxrec` form since a named syntax declaration is "obviously" recursive in the same way a function declaration is recursive:

```js
syntax m (ctx) {
  return #`m` // infinite expansion
}
```

**Comment by gabejohnson** [comment]
So for default imports we would have:
```js
// export.js
export default syntax(ctx) {

}
//--------------------------
// import.js
import whateverNameIWant from './from.js';
```

So in the importing file I can alias the macro name?

**Comment by disnet** [comment]
> In the importing file I can now alias the macro. Is that what we want?
 
Yeah, absolutely. It would be surprising if we didn't support this.

**Comment by disnet** [comment]
Heh, just realized that flow has a similar problem. You can export a type via `export type F = string` but you can't export a default type.
